node_modules/col-activities/lib/api.js

```diff
@@ -14,16 +14,99 @@
 
 /* Activities */
 
 /**
+ * Get the activities for a course that contributed points to the engagement index
+ *
+ * @param  {Context}          ctx                             Standard context containing the current user and the current course
+ * @param  {Function}         callback                        Standard callback function
+ * @param  {Object}           callback.err                    An error that occurred, if any
+ * @param  {Activity[]}       callback.activities             The activities for the course
+ * @api private
+ */
+var getActivities = function(ctx, callback) {
+  // Get the activities from the DB
+  var options = {
+    'where': {
+      'course_id': ctx.course.id
+    },
+    'include': {
+      'model': DB.User,
+      'attributes': UserConstants.BASIC_USER_FIELDS
+    }
+  };
+
+  DB.Activity.findAll(options).complete(function(err, activities) {
+    if (err) {
+      log.error({'err': err, 'course': ctx.course.id}, 'Failed to get the activities');
+      return callback({'code': 500, 'msg': err.message});
+    }
+
+    return callback(null, activities);
+  });
+};
+
+/**
+ * Export the activities for a course as a CSV file
+ *
+ * @param  {Context}          ctx                             Standard context containing the current user and the current course
+ * @param  {Function}         callback                        Standard callback function
+ * @param  {Object}           callback.err                    An error that occurred, if any
+ * @param  {String}           callback.activities             A CSV export of the activities for the course
+ */
+var exportActivities = module.exports.exportActivities = function(ctx, callback) {
+  // Only administrators are able to get the activities for a course
+  if (!ctx.user.is_admin) {
+    log.error({'id': ctx.user.id}, 'Unauthorized to get the activities');
+    return callback({'code': 401, 'msg': 'Unauthorized to get the activities'});
+  }
+
+  // Get the activity type configuration for the course
+  getActivityTypeConfiguration(ctx.course.id, function(err, configuration) {
+    if (err) {
+      return callback(err);
+    }
+
+    // Get the activities for the course
+    getActivities(ctx, function(err, activities) {
+      if (err) {
+        return callback(err);
+      }
+
+      // Format the activities into a CSV file
+      var formattedActivities = [];
+      var totalScores = {};
+      _.each(activities, function(activity) {
+        var formattedActivity = {
+          'user_id': activity.user_id,
+          'user_name': activity.user.canvas_full_name,
+          'action': activity.type,
+          'date': activity.created_at
+        };
+        formattedActivity.score = _.findWhere(configuration, {'type': activity.type}).points;
+        // Add the running score for the user
+        totalScores[activity.user_id] = (totalScores[activity.user_id] || 0) + formattedActivity.score;
+        formattedActivity.total = totalScores[activity.user_id];
```

Comment:
Can we call this property `runningTotal` and default it to 0 when instantiating the `formattedActivity`?